py-polars/tests/unit/operations/test_inefficient_apply.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+from typing import Any, Callable
+from warnings import catch_warnings, simplefilter
+
+import numpy as np
+import pytest
+
+import polars as pl
+from polars.exceptions import PolarsInefficientApplyWarning
+from polars.utils.udfs import (
+    _get_bytecode_ops,
+    _is_inefficient,
+    _rewrite_as_polars_expr,
+    _simple_signature,
+)
+
+
+def _get_suggestion(
+    func: Callable[[Any], Any], col: str, apply_target: str
+) -> str | None:
+    return _rewrite_as_polars_expr(_get_bytecode_ops(func), col, apply_target)
+
+
+@pytest.mark.parametrize(
+    "func",
+    [
+        np.sin,
+        lambda x: np.sin(x),
+        lambda x, y: x + y,
+    ],
+)
```

Comment:
can we please also check:
- what happens if variables from outside the lambda's scope are used?
- what happens if `x` is indexed (e.g. `x[0]`)

Reply:
> * what happens if variables from outside the lambda's scope are used?

That would generate bytecode that includes `LOAD_GLOBAL` or `LOAD_DEREF` ops, so won't be allowed.

> * what happens if `x` is indexed (e.g. `x[0]`)

That would use `BINARY_SUBSCR` which also isn't currently allowed (though we'll need it later for DataFrame apply).

I'll add some bonus coverage ;)